nightly/PRs: IAR build errors with an internal error

<!--

   ************************************** WARNING **************************************

   The ciarcom bot parses this header automatically. Any deviation from the 
   template may cause the bot to automatically correct this header or may result in a 
   warning message, requesting updates.

   Please ensure all sections of the template below are filled in and no changes 
   are made to the template format. Only bugs should be raised here as issues. 
   Questions or enhancements should instead be raised on our forums:
   https://forums.mbed.com/ .

   *************************************************************************************

-->

### Description of defect

<!--
    Add detailed description of what you are reporting.
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html
-->

This currently affects PRs and it is also in the latest nightly (was not in the day before). Something has changed in CI or example?

This is the example failing `mbed-os-example-socket-stats` - cc @ARMmbed/mbed-os-core 

```
[ERROR] Internal error: [Front end]: assertion failed at: "/home/jenkins_ci01/slave_top/workspace/ArmReleaseBuild/core/translator/compiler_core/src/parser/edg/il.c", line 23626


Fatal error detected, aborting.

[mbed] Working path "/builds/ws/mbed-os-ci_build-IAR@3/examples/mbed-os-example-socket-stats" (program)
```

As nightly and also PRs (one is using master and one release branch), this does not indicate it is in the example itself but rather a change in CI within the last 15 hours or so?

cc @ARMmbed/mbed-os-test 

#### Target(s) affected by this defect ?

K64F

#### Toolchain(s) (name and version) displaying this defect ?

IAR

#### What version of Mbed-os are you using (tag or sha) ?
<!--
    For a released version please provide the release tag (this can be found as per the instructions below)

    mbed-os version can be found in /platform/mbed_version.h. The tag can be reconstructed as follows:
    mbed-os-MBED_MAJOR_VERSION.MBED_MINOR_VERSION.MBED_PATCH_VERSION
 
    Master branch is indicated by 'mbed-os-99.99.99
    
    For an issue found on Master please provide the sha being used.
-->

mbed-os-99.99.99

#### What version(s) of tools are you using. List all that apply (E.g. mbed-cli)

CI related

#### How is this defect reproduced ? 

I am testing locally , will update

--- a/main.cpp
+++ b/main.cpp
@@ -28,7 +28,7 @@ EventFlags threadFlag;
 
 void print_socket_stats()
 {
-    mbed_stats_socket_t stats[MBED_CONF_NSAPI_SOCKET_STATS_MAX_COUNT] = {0};
+    static mbed_stats_socket_t stats[MBED_CONF_NSAPI_SOCKET_STATS_MAX_COUNT];
     int iteration = 0;
 
     while (COMPLETED_FLAG != threadFlag.get()) {

